py7zr: init at 0.22.0; pyppmd: init at 1.1.1; pybcj: init at 1.0.3; multivolumefile: init at 0.2.3; inflate64: init at 1.0.1

pitkling opened the pull request:
Continues the work done by @poperigby (in PR #365111) and @ByteSudoer (in PR #324293).

* closes #365111
* closes #324293
* closes #324207

## Commit Explanation:
* The first five commits are just a cleanup of the commit history of PR #365111. Otherwise **no** changes.
* Commit six adds me as a maintainer to the related `pyzstd`. It was originally part of PR #365111 but has been added in another PR #381975 by @MattSturgeon that has been merged.
* The remaining commits are new changes by me. The commits are temporary to ease reviews of anyone that previously reviewed PR #365111. I'll squash them into the corresponding package-related commits later on.
    * They mostly just activate standard python checks (as suggested by @MattSturgeon in [this comment](https://github.com/NixOS/nixpkgs/pull/365111#issuecomment-2674996025)).
    * There are also two substitutes to fix minor problems in the upstream packages:
        * an inaccessible mirror in test of `py7zr` (upstream PR: miurahr/py7zr#637)
        * typo in `pyproject.toml` of `multivolumefile` ([upstream PR](https://codeberg.org/miurahr/multivolume/pulls/9))

## Potential Reviewers
* @poperigby, @ByteSudoer: initiated previous PRs for these packages
* @MattSturgeon: cherry picked the related `pyzstd` from the original PRs
* @natsukium: gave a lot of helpful feedback on the cherry-picked `pyzstd` in PR #381975

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

pitkling committed:
inflate64: init at 1.0.1

Co-authored-by: ByteSudoer <ByteSudoer@users.noreply.github.com>
Co-authored-by: PopeRigby <poperigby@mailbox.org>

pitkling committed:
multivolumefile: init at 0.2.3

Co-authored-by: ByteSudoer <ByteSudoer@users.noreply.github.com>
Co-authored-by: Matt Sturgeon <matt@sturgeon.me.uk>
Co-authored-by: PopeRigby <poperigby@mailbox.org>

pitkling committed:
pybcj: init at 1.0.3

Co-authored-by: ByteSudoer <ByteSudoer@users.noreply.github.com>
Co-authored-by: PopeRigby <poperigby@mailbox.org>

pitkling committed:
pyppmd: init at 1.1.1

Co-authored-by: ByteSudoer <ByteSudoer@users.noreply.github.com>
Co-authored-by: PopeRigby <poperigby@mailbox.org>

pitkling committed:
py7zr: init at 0.22.0

Co-authored-by: ByteSudoer <ByteSudoer@users.noreply.github.com>
Co-authored-by: Matt Sturgeon <matt@sturgeon.me.uk>
Co-authored-by: PopeRigby <poperigby@mailbox.org>

pitkling committed:
pyzstd: add pitkling as maintainer

pitkling commented:
@poperigby: I kept you as a maintainer (as by your original PR); let me know if that's ok.

MattSturgeon reviewed (commented):
Thanks for continuing this work. This is looking good!

I've left some comments for a few minor things I think could be cleaned up a little.

pitkling commented on pkgs/development/python-modules/multivolumefile/fix-pyproject-toml-typo.patch:
```diff
@@ -0,0 +1,13 @@
+ build-backend = "setuptools.build_meta"
+ 
+-[tools.setuptools_scm]
++[tool.setuptools_scm]
```
Thanks, didn't even think of `substituteInPlace`; it's much cleaner I think 🙂. Done.

pitkling commented on pkgs/development/python-modules/py7zr/fix-qtbase-url.patch:
```diff
@@ -0,0 +1,13 @@
+     (
+         "qtbase.7z",
+-        "http://qt.mirrors.tds.net/qt/online/qtsdkrepository/"
++        "http://download.qt.io/online/qtsdkrepository/"
```
Done. (And yes, `https` works, I changed it accordingly.)

pitkling commented on pkgs/top-level/python-packages.nix:
```diff
@@ -6380,6 +6380,8 @@ self: super: with self; {
 
   infinity = callPackage ../development/python-modules/infinity { };
 
+  inflate64 = callPackage ../development/python-modules/inflate64 { };
+
```
Done.

MattSturgeon commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384329`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 20 packages built:</summary>
  <ul>
    <li>py7zr (python312Packages.py7zr)</li>
    <li>py7zr.dist (python312Packages.py7zr.dist)</li>
    <li>python312Packages.inflate64</li>
    <li>python312Packages.inflate64.dist</li>
    <li>python312Packages.multivolumefile</li>
    <li>python312Packages.multivolumefile.dist</li>
    <li>python312Packages.pybcj</li>
    <li>python312Packages.pybcj.dist</li>
    <li>python312Packages.pyppmd</li>
    <li>python312Packages.pyppmd.dist</li>
    <li>python313Packages.inflate64</li>
    <li>python313Packages.inflate64.dist</li>
    <li>python313Packages.multivolumefile</li>
    <li>python313Packages.multivolumefile.dist</li>
    <li>python313Packages.py7zr</li>
    <li>python313Packages.py7zr.dist</li>
    <li>python313Packages.pybcj</li>
    <li>python313Packages.pybcj.dist</li>
    <li>python313Packages.pyppmd</li>
    <li>python313Packages.pyppmd.dist</li>
  </ul>
</details>

MattSturgeon reviewed (approved):
Diff lgtm, once the tmp commits are squashed.

pitkling commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384329`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 20 packages built:</summary>
  <ul>
    <li>py7zr (python312Packages.py7zr)</li>
    <li>py7zr.dist (python312Packages.py7zr.dist)</li>
    <li>python312Packages.inflate64</li>
    <li>python312Packages.inflate64.dist</li>
    <li>python312Packages.multivolumefile</li>
    <li>python312Packages.multivolumefile.dist</li>
    <li>python312Packages.pybcj</li>
    <li>python312Packages.pybcj.dist</li>
    <li>python312Packages.pyppmd</li>
    <li>python312Packages.pyppmd.dist</li>
    <li>python313Packages.inflate64</li>
    <li>python313Packages.inflate64.dist</li>
    <li>python313Packages.multivolumefile</li>
    <li>python313Packages.multivolumefile.dist</li>
    <li>python313Packages.py7zr</li>
    <li>python313Packages.py7zr.dist</li>
    <li>python313Packages.pybcj</li>
    <li>python313Packages.pybcj.dist</li>
    <li>python313Packages.pyppmd</li>
    <li>python313Packages.pyppmd.dist</li>
  </ul>
</details>

pitkling commented:
> @pitkling can you please clean the git history?

@GaetanLepage: Just cleaned it up (and added two comments with references to the upstream PRs I created for the two substitutions).

SuperSandro2000 commented on pkgs/development/python-modules/inflate64/default.nix:
```diff
@@ -0,0 +1,47 @@
+buildPythonPackage rec {
+  pname = "inflate64";
+  version = "1.0.1";
+
```
```suggestion
```

MattSturgeon replied:
I'm not convinced these minor stylistic preferences are worth the friction of requesting changes.

Seems to go against the principles laid out in [How to review pull requests](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#how-to-review-pull-requests) and [A reviewer requested a bunch of insubstantial changes on my PR](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#a-reviewer-requested-a-bunch-of-insubstantial-changes-on-my-pr), though maybe my interpretation is off.

Maybe consider marking such minor feedback as a nitpick, or explain why you think it is important? :slightly_smiling_face:

SuperSandro2000 commented on pkgs/development/python-modules/inflate64/default.nix:
```diff
@@ -0,0 +1,47 @@
+    domain = "codeberg.org";
+    owner = "miurahr";
+    repo = "inflate64";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

SuperSandro2000 commented on pkgs/development/python-modules/multivolumefile/default.nix:
```diff
@@ -0,0 +1,57 @@
+buildPythonPackage rec {
+  pname = "multivolumefile";
+  version = "0.2.3";
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/multivolumefile/default.nix:
```diff
@@ -0,0 +1,57 @@
+    domain = "codeberg.org";
+    owner = "miurahr";
+    repo = "multivolume";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

SuperSandro2000 commented on pkgs/development/python-modules/multivolumefile/default.nix:
```diff
@@ -0,0 +1,57 @@
+
+  postPatch = ''
+    # Fix typo: `tools` -> `tool`
+    # (upstream PR: https://codeberg.org/miurahr/multivolume/pulls/9)
```
```suggestion
    # upstream PR: https://codeberg.org/miurahr/multivolume/pulls/9
```

SuperSandro2000 commented on pkgs/development/python-modules/py7zr/default.nix:
```diff
@@ -0,0 +1,81 @@
+
+  postPatch = ''
+    # Replace inaccessible mirror (qt.mirrors.tds.net):
+    # (upstream PR: https://github.com/miurahr/py7zr/pull/637)
```
```suggestion
    # upstream PR: https://github.com/miurahr/py7zr/pull/637
```

SuperSandro2000 commented on pkgs/development/python-modules/py7zr/default.nix:
```diff
@@ -0,0 +1,81 @@
+    # Replace inaccessible mirror (qt.mirrors.tds.net):
+    # (upstream PR: https://github.com/miurahr/py7zr/pull/637)
+    substituteInPlace tests/test_concurrent.py \
+        --replace-fail 'http://qt.mirrors.tds.net/qt/' \
+                       'https://download.qt.io/'
```
```suggestion
      --replace-fail 'http://qt.mirrors.tds.net/qt/' 'https://download.qt.io/'
```

pitkling replied:
I slightly preferred the previous indentation myself. But then I'm not too familiar with how substitutes are typically formatted in nixpkgs, so I implemented it as suggested :).

SuperSandro2000 commented on pkgs/development/python-modules/pybcj/default.nix:
```diff
@@ -0,0 +1,48 @@
+buildPythonPackage rec {
+  pname = "pybcj";
+  version = "1.0.3";
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/pybcj/default.nix:
```diff
@@ -0,0 +1,48 @@
+    domain = "codeberg.org";
+    owner = "miurahr";
+    repo = "pybcj";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

SuperSandro2000 commented on pkgs/development/python-modules/pyppmd/default.nix:
```diff
@@ -0,0 +1,52 @@
+buildPythonPackage rec {
+  pname = "pyppmd";
+  version = "1.1.1";
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/pyppmd/default.nix:
```diff
@@ -0,0 +1,52 @@
+    domain = "codeberg.org";
+    owner = "miurahr";
+    repo = "pyppmd";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

pitkling commented:
@SuperSandro2000: Thanks for the formatting fixes, I just implemented them.

pitkling commented on pkgs/development/python-modules/inflate64/default.nix:
```diff
@@ -0,0 +1,46 @@
+  lib,
+  pytestCheckHook,
+  setuptools,
+  setuptools-scm,
```
Thanks, done for all packages added by the PR.

Actually, I always wondered about which order is typically used for the inputs but couldn't find any information on it. Out of curiosity, since I apparently also overlooked the (very helpful!) "Practical Contributing Advice" linked in [another comment](https://github.com/NixOS/nixpkgs/pull/384329#discussion_r1966873348) by @MattSturgeon: is this (and similar best practices) documented somewhere?

GaetanLepage reviewed (commented):
This looks very close.

pitkling commented on pkgs/development/python-modules/multivolumefile/default.nix:
```diff
@@ -0,0 +1,56 @@
+    # upstream PR: https://codeberg.org/miurahr/multivolume/pulls/9
+    substituteInPlace pyproject.toml \
+      --replace-fail 'tools.setuptools_scm' 'tool.setuptools_scm'
+  '';
```
Done, and thanks for the explanation!

GaetanLepage reviewed (approved):
LGTM! Thanks for your patience @pitkling
